modules/tinymce/src/core/test/ts/browser/selection/NormalizeRangeTest.ts

```diff
@@ -1,314 +1,340 @@
 import * as NormalizeRange from 'tinymce/core/selection/NormalizeRange';
 import * as Zwsp from 'tinymce/core/text/Zwsp';
-import ViewBlock from '../../module/test/ViewBlock';
+import * as ViewBlock from '../../module/test/ViewBlock';
 
-UnitTest.asynctest('browser.tinymce.core.selection.NormalizeRangeTest', (success, failure) => {
-  const viewBlock = ViewBlock();
+describe('browser.tinymce.core.selection.NormalizeRangeTest', () => {
+  const viewBlock = ViewBlock.bddSetup();
 
-  const assertRange = (root, range, startPath, startOffset, endPath, endOffset) => {
+  const assertRange = (root: Node, range: Optional<Range>, startPath: number[], startOffset: number, endPath: number[], endOffset: number) => {
     const sc = Hierarchy.follow(SugarElement.fromDom(root), startPath).getOrDie();
     const ec = Hierarchy.follow(SugarElement.fromDom(root), endPath).getOrDie();
     const actualRange = range.getOrDie('Should be some');
 
     Assertions.assertDomEq('Should be expected start container', sc, SugarElement.fromDom(actualRange.startContainer));
-    Assertions.assertEq('Should be expected start offset', startOffset, actualRange.startOffset);
+    assert.equal(actualRange.startOffset, startOffset, 'Should be expected start offset');
     Assertions.assertDomEq('Should be expected end container', ec, SugarElement.fromDom(actualRange.endContainer));
-    Assertions.assertEq('Should be expected end offset', endOffset, actualRange.endOffset);
+    assert.equal(actualRange.endOffset, endOffset, 'Should be expected end offset');
   };
 
-  const cSetHtml = (html) => {
-    return Chain.op(() => {
-      viewBlock.update(html);
-    });
+  const setHtml = viewBlock.update;
+
+  const normalizeRange = (startPath: number[], startOffset: number, endPath: number[], endOffset: number) => {
+    const sc = Hierarchy.follow(SugarElement.fromDom(viewBlock.get()), startPath).getOrDie();
+    const ec = Hierarchy.follow(SugarElement.fromDom(viewBlock.get()), endPath).getOrDie();
+    const rng = document.createRange();
+
+    rng.setStart(sc.dom, startOffset);
+    rng.setEnd(ec.dom, endOffset);
+
+    return NormalizeRange.normalize(DOMUtils(document, { root_element: viewBlock.get() }), rng);
   };
 
-  const cNormalizeRange = (startPath, startOffset, endPath, endOffset) => {
-    return Chain.mapper((viewBlock: any) => {
-      const sc = Hierarchy.follow(SugarElement.fromDom(viewBlock.get()), startPath).getOrDie();
-      const ec = Hierarchy.follow(SugarElement.fromDom(viewBlock.get()), endPath).getOrDie();
-      const rng = document.createRange();
+  const assertRangeNone = (range: Optional<Range>) => {
+    assert.isTrue(range.isNone(), 'Should be none');
+  };
 
-      rng.setStart(sc.dom, startOffset);
-      rng.setEnd(ec.dom, endOffset);
+  context('Non normalize non collapsed selections', () => {
+    it('Should not normalize on indexed selected at root level', () => {
+      setHtml('<input>');
+      const range = normalizeRange( [], 0, [], 1);
```

Comment:
```suggestion
      const range = normalizeRange([], 0, [], 1);
```

Reply:
Hmm, old formatting issue it seems. We possibly should look at our linting rules further... Anyways, fixed for now.